File: backend-java/src/main/java/br/com/karatedopi/controllers/dtos/RegistrationFormInputDTO.java
package br.com.karatedopi.controllers.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;
import lombok.Builder;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RegistrationFormInputDTO {

    @JsonProperty("user")
    private UserInputDTO userInputDTO;

    @JsonProperty("profile")
    private ProfileInputDTO profileInputDTO;

    @JsonProperty("address")
    private AddressDTO addressDTO;
}

File: frontend-angular/src/app/common/registration.form.input.dto.ts
import { AddressDto } from "./address.dto";
import { ProfileInputDto } from "./profile.input.dto";
import { UserInputDto } from "./user.input.dto";

export class RegistrationFormInputDto {
    constructor(
        public user: UserInputDto,
        public profile: ProfileInputDto,
        public address: AddressDto
    ) {}
}
